contrail_api_cli_extra/clean/fip.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+from contrail_api_cli.utils import printo
+from contrail_api_cli.exceptions import ResourceNotFound, CollectionNotFound
+
+from ..utils import server_type, CassandraCommand, CheckCommand, PathCommand
+from refs import CleanRefs
+
+class CleanFipRefs(CassandraCommand, CheckCommand, PathCommand):
+    """Clean broken references related to floating IPs.
+
+    Broken refs related to FIP can be found with gremlin (floating-ip-pool <->
+    floating-ip).
+
+    The command will take all existing floating-ip-pool and checks the existence
+    of each floating-ip it points to and removes the reference if that latter
+    does not exist.
+
+    You can run the command by putting the floating-ip-pool path to be more
+    precise or you can run it without any resources path, in that case it will
+    iterates over all floating-ip-pool.
+
+    """
+    description = "Clean broken references related to floating IPs"
+
+    @property
+    def resource_type(self):
+        return "floating-ip-pool"
+
+    def _clean_refs(self, pool, fip, **kwargs):
+        try:
+            fip.fetch()
+        except ResourceNotFound:
+            CleanRefs(ref_type="children",
+                      target_type="floating_ip",
+                      paths=[pool.uuid, fip.uuid],
+                      **kwargs)
+
+    def __call__(self, **kwargs):
+        super(CleanFipRefs, self).__call__(**kwargs)
+        for pool in self.resources:
+            try:
+                pool.fetch()
+                for fip in pool.children:
```

Comment:
Better to be more explicit and do `pool.children.floating_ip`

Reply:
Also, it would ne nice to be able to pass (optionaly) a fixed list of floating-ip uuids to the command, so that we can use this command in gremlin-fsck (as with gremlin we already know which FIPs are missing)